File: server/BarAPP/src/main/java/org/pbarreiro/barapp/model/Producto.java
package org.pbarreiro.barapp.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

@Entity
@Table(name = "productos")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Producto {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String nombre;

    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal precio;

    @Column(columnDefinition = "int default 0")
    private Integer stock;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "categoria_id")
    private Categoria categoria;

    @Column(name = "imagen_url")
    private String imagenUrl;

    @Column(columnDefinition = "boolean default true")
    private Boolean activo;
}

File: mobile_client/BarAPP/composeApp/src/commonMain/kotlin/org/pabarreiro/barapp/domain/model/Producto.kt
package org.pabarreiro.barapp.domain.model

import kotlinx.serialization.Serializable

@Serializable
data class Producto(
    val id: Long,
    val nombre: String,
    val precio: Double,
    val stock: Int,
    val categoriaId: Long?,
    val imagenUrl: String?,
    val activo: Boolean
)
